### PR TITLE
Update Selectors-Advanced/0010

### LINKED ISSUE #51: Update selectors-advanced/0010 - `a:not(:link)` to `a:visited`
In `selectors-advanced/0010`, we have:

```css
a:not(:link) {
  color: red;
}
```
```css
a:visited{color:red}
```

This is good, however, there should be a test to ensure this does not apply to elements that cannot use `:visited`. 

* * *

For example, classes should not be minified, because the minifier cannot ensure they will be exclusively used by links:

```css
.foo:not(:link){color:red}
.foo:visited{color:red}
```

```html
<div class="foo">Text</div>
<a class="foo" href="#">Text</a>
```

* * *

And HTML elements that do not support `:visited` must not be minified.

```css
/* Selector matches ALL H1 elements */
h1:not(:link){color:red}

/* Selector matches NO H1 elements */
h1:visited{color:red}
```

* * *

At first, I thought `[href]` would be safe, but arbitrary elements could have an `href` attribute set, even if non-standard (`<div href="foo">`).

However, if the selector had a qualifying element that we know supports this minification, then it would be fine.

```css
[href]:not(:link),
[href="#"]:not(:link),
a[href]:not(:link),
area[href="#"]:not(:link) {
  color: red;
}
```
```css
[href]:not(:link),[href="#"]:not(:link),a[href]:visited,area[href="#"]:visited{color:red}
```

* * *

Should the source/expected for 0010 be updated to include more examples, and the README updated to explain the edge cases, or should the edge cases be separate tests?

**Combined example:**

```css
a:not(:link),
area:not(:link),
link:not(:link),
.foo:not(:link),
h1:not(:link),
[href]:not(:link),
[href="#"]:not(:link),
a[href]:not(:link),
area[href="#"]:not(:link) {
  color: red;
}
```
```css
a:visited,area:visited,link:visited,.foo:not(:link),h1:not(:link),[href]:not(:link),[href="#"]:not(:link),a[href]:visited,area[href="#"]:visited{color:red}
```

README update:

```
On hyperlink elements (`<a>`, `<area>`, `<link>` with href) `:link` and
`:visited` are a complete partition. `a:not(:link)` can be replaced with
the shorter `a:visited`. However, this should not apply to non-hyperlink
elements (`<div>`, `<h1>`, etc), and should not apply to non-tag selectors,
for example, a class could be applied to both `<a class="x">` and `<div class="x">`.
HREF attribute selectors should only receive this minification if they have
a qualifying element of `a`, `area`, or `link`.
```

### PR DESCRIPTION
**Pull request by TheJaredWilcurt**
This closes #51. I'll let you handle the mathml/svg stuff mentioned in that issue in a different PR, I'm not sure what you want for that.

No changes to test suite scores, all minifiers were previously failing this test and still are now that it handles additional edgecases.

### CODE REVIEW

File: tests/selectors-advanced/0010/source.css [hunk 1]
```diff
@@ -1,3 +1,11 @@
-a:not(:link) {
+a:not(:link),
+area:not(:link),
+link:not(:link),
+.foo:not(:link),
+h1:not(:link),
+[href]:not(:link),
+[href="#"]:not(:link),
+a[href]:not(:link),
+area[href="#"]:not(:link) {
```
Comment (keithamus) [inline tests/selectors-advanced/0010/source.css]: While here maybe it's worth also adding `:where(:not(:link))` (which should be `:where(:visited)` some of the time).

**Review by keithamus — APPROVED** [review]
Generally LGTM but what do you think about...

File: tests/selectors-advanced/0010/source.css (hunk 1 above)
Comment (TheJaredWilcurt) [reply tests/selectors-advanced/0010/source.css]: `:where(:not(:link))` would target any non-link on the page, and any link that is visited (almost every element).

`:where(:visited)` would only target links that have been visited and no other elements. So these are two very different outcomes. A minifier should not change one to the other. I'll add it to the test.

Comment (keithamus) [reply tests/selectors-advanced/0010/source.css]: Right but `a:where(:not(:link))` is `a:where(:visited)` AIUI.

Comment (TheJaredWilcurt) [reply tests/selectors-advanced/0010/source.css]: Correct you are. Tests amended.

**Review by keithamus — APPROVED** [review]
Nice this is looking great!
